### PR TITLE
Fix Trilinos mat-mat multiplication

### LINKED ISSUE #2531: TrilinosWrappers::SparseMatrix.mmult() not working correctly
There seems to be a problem with the TrilinosWrappers::SparseMatrix mmult() function, e.g, multiplying matrices with sizes `A = 4 x 3` and `B = 3 x 5`

using `A.mmult(AB, B)` yields `AB = 4 x 4` instead of `4 x 5`.

Working example:
[mmult_Error.cpp.zip](https://github.com/dealii/dealii/files/228945/mmult_Error.cpp.zip)

### PR DESCRIPTION
**Pull request by kronbichler**
Fixes #2531.

### CODE REVIEW

File: source/lac/trilinos_sparse_matrix.cc [hunk 1]
```diff
@@ -2295,7 +2295,9 @@ namespace TrilinosWrappers
       // matrix that we got as a result.
       Epetra_CrsMatrix *C_mat;
       ML_Operator2EpetraCrsMatrix(C_, C_mat);
-      C_mat->FillComplete();
+      C_mat->FillComplete(mod_B->DomainMap(),
+                          transpose_left ? inputleft.trilinos_matrix().DomainMap() :
+                          inputleft.trilinos_matrix().RangeMap());
```
Comment (bangerth) [inline source/lac/trilinos_sparse_matrix.cc]: Nice! This is really all? ;-)

Please break the `?` case onto a line of its own to make it symmetric with the `:` case.

File: tests/trilinos/sparse_matrix_mmult_01.cc [hunk 2]
```diff
@@ -0,0 +1,65 @@
+// ---------------------------------------------------------------------
+//
+// Copyright (C) 2016 - 2016 by the deal.II authors
```
Comment (bangerth) [inline tests/trilinos/sparse_matrix_mmult_01.cc]: Not strictly necessary to have a whole range here :-)

**Comment by bangerth** [comment]
OK to merge with the minor text changes and if you add an entry to `changes.h` (maybe with credit also to @jodlbauer). Thanks for the quick fix!

(Oh, you also have to fix the indentation.)

**Comment by kronbichler** [comment]
@jodlbauer would you mind sharing your full name for the changelog? Thanks!

**Comment by jodlbauer** [comment]
Thanks for fixing!

Daniel Jodlbauer

**Comment by tjhei** [comment]
Would you mind adding the Tmmult case to the test?

**Comment by kronbichler** [comment]
I added a test for the transpose case, thanks for the remark.

**Pull request merged**